File: app/src/main/java/org/wikipedia/page/GeoMarshaller.java
package org.wikipedia.page;

import android.location.Location;
import android.support.annotation.Nullable;

import org.json.JSONException;
import org.json.JSONObject;

public final class GeoMarshaller {
    @Nullable
    public static String marshal(@Nullable Location object) {
        if (object == null) {
            return null;
        }

        JSONObject jsonObj = new JSONObject();
        try {
            jsonObj.put(GeoUnmarshaller.LATITUDE, object.getLatitude());
            jsonObj.put(GeoUnmarshaller.LONGITUDE, object.getLongitude());
        } catch (JSONException e) {
            throw new RuntimeException(e);
        }
        return jsonObj.toString();
    }

    private GeoMarshaller() { }
}

File: app/src/main/java/org/wikipedia/page/GeoMarshaller.kt
package org.wikipedia.page

import android.location.Location
import org.json.JSONException
import org.json.JSONObject

object GeoMarshaller {
    @JvmStatic
    fun marshal(obj: Location?): String? {
        return obj?.let {
            JSONObject().runCatching {
                put(GeoUnmarshaller.LATITUDE, it.latitude)
                put(GeoUnmarshaller.LONGITUDE, it.longitude)
            }.onFailure {
                throw RuntimeException(it as JSONException)
            }.toString()
        }
    }
}
